core/src/main/java/nu/yona/server/crypto/seckey/SecretKeyUtil.java

```diff
@@ -0,0 +1,196 @@
+ * v. 2.0. If a copy of the MPL was not distributed with this file, You can obtain one at https://mozilla.org/MPL/2.0/.
+ *******************************************************************************/
+package nu.yona.server.crypto.seckey;
+
+import java.nio.ByteBuffer;
+import java.nio.charset.StandardCharsets;
+import java.security.NoSuchAlgorithmException;
+import java.util.Arrays;
+import java.util.UUID;
+
+import javax.crypto.BadPaddingException;
+import javax.crypto.Cipher;
+import javax.crypto.IllegalBlockSizeException;
+import javax.crypto.KeyGenerator;
+import javax.crypto.SecretKey;
+import javax.crypto.ShortBufferException;
+import javax.crypto.spec.SecretKeySpec;
+
+import nu.yona.server.crypto.CryptoException;
+import nu.yona.server.crypto.CryptoUtil;
+
+public class SecretKeyUtil
+{
+	public static final int INITIALIZATION_VECTOR_LENGTH = 16;
+
+	private SecretKeyUtil()
+	{
+		// No instances
+	}
+
+	public static byte[] encryptUuid(UUID plaintext)
+	{
+		if (plaintext == null)
+		{
+			return null;
+		}
+		ByteBuffer bb = ByteBuffer.wrap(new byte[16]);
+		bb.putLong(plaintext.getMostSignificantBits());
+		bb.putLong(plaintext.getLeastSignificantBits());
+		return encryptBytes(bb.array());
+	}
+
+	public static byte[] encryptString(String plaintext)
+	{
+		if (plaintext == null)
+		{
+			return null;
+		}
+		return encryptBytes(plaintext.toString().getBytes(StandardCharsets.UTF_8));
+	}
+
+	public static byte[] encryptLong(Long plaintext)
+	{
+		if (plaintext == null)
+		{
+			return null;
+		}
+		ByteBuffer bb = ByteBuffer.wrap(new byte[16]);
+		bb.putLong(plaintext);
+		return encryptBytes(bb.array());
+	}
+
+	public static byte[] encryptBytes(byte[] plaintext)
+	{
+		if (plaintext == null)
+		{
+			return null;
+		}
+		return CryptoSession.getCurrent().encrypt(plaintext);
+	}
+
+	public static UUID decryptUuid(byte[] ciphertext)
+	{
+		if (ciphertext == null)
+		{
+			return null;
+		}
+		ByteBuffer bb = ByteBuffer.wrap(decryptBytes(ciphertext));
+		long firstLong = bb.getLong();
+		long secondLong = bb.getLong();
+		return new UUID(firstLong, secondLong);
+	}
+
+	public static String decryptString(byte[] ciphertext)
+	{
+		if (ciphertext == null)
+		{
+			return null;
+		}
+		return new String(decryptBytes(ciphertext), StandardCharsets.UTF_8);
+	}
+
+	public static Long decryptLong(byte[] ciphertext)
+	{
+		if (ciphertext == null)
+		{
+			return null;
+		}
+		ByteBuffer bb = ByteBuffer.wrap(decryptBytes(ciphertext));
+		return bb.getLong();
+	}
+
+	public static byte[] decryptBytes(byte[] ciphertext)
+	{
+		if (ciphertext == null)
+		{
+			return null;
+		}
+		return CryptoSession.getCurrent().decrypt(ciphertext);
+	}
+
+	/**
+	 * Encrypts the given plaintext bytes.
+	 * 
+	 * @param plaintext the bytes to be encrypted
+	 * @return the encrypted bytes, with the crypto variant number prepended to it.
+	 */
+	public static byte[] encrypt(byte cryptoVariantNumber, Cipher cipher, byte[] plaintext)
```

Comment:
This method should certainly be in the shared `CryptoUtil` class, because it is used both by public key and secret key encryption. The cipher is passed.